File: android/app/src/main/java/com/aiusage/mobile/sync/SnapshotStatus.kt
package com.aiusage.mobile.sync

enum class SnapshotStatus {
    NotLinked,
    Fresh,
    Stale,
    Offline,
    ProviderError,
    Revoked
}

fun resolveSnapshotStatus(
    ageSeconds: Long?,
    hasProviderError: Boolean,
    revoked: Boolean = false
): SnapshotStatus {
    if (revoked) return SnapshotStatus.Revoked
    if (ageSeconds == null) return SnapshotStatus.NotLinked
    if (ageSeconds >= 24 * 60 * 60) return SnapshotStatus.Offline
    if (hasProviderError) return SnapshotStatus.ProviderError
    if (ageSeconds >= 15 * 60) return SnapshotStatus.Stale
    return SnapshotStatus.Fresh
}

File: ios/AIQuotaMobile/SnapshotStatus.swift
import Foundation

enum SnapshotStatus: String, Codable {
    case notLinked
    case fresh
    case stale
    case offline
    case providerError
    case revoked

    static func resolve(uploadedAt: Date?, providerStatuses: [ProviderStatus], now: Date = Date()) -> SnapshotStatus {
        guard let uploadedAt else {
            return .notLinked
        }

        let age = now.timeIntervalSince(uploadedAt)
        if age >= 24 * 60 * 60 {
            return .offline
        }
        if providerStatuses.contains(.error) {
            return .providerError
        }
        if age >= 15 * 60 {
            return .stale
        }
        return .fresh
    }
}

